### ISSUE TITLE
Debt: Consolidate mod processing in addRecordSRV/addRecord

### ISSUE THREAD

**Issue by tlimoncelli** [StackExchange/dnscontrol#157]
There is a lot of repeated code in addRecordSRV/addRecord.  It should be collapsed into a helper function.

**Comment by whs** [comment]
I could take on this.

I'm thinking of something like:

```js
var CNAME = recordBuilder({
    type: 'CNAME',
});
```

which should take care of the (name, target) type.

For special cases like SRV a hook could be added

```js
var SRV = recordBuilder({
    type: 'SRV',
    args: [
		['name', _.isString], // second argument can be omitted/null if validation is not needed
		['priority', _.isNumber],
		['weight', _.isNumber],
		['port', _.isNumber],
		['target', _.isString],
	],
    transform: function(record, args, modifiers){
		// record is {type: 'SRV', ttl:d.defaultTTL, meta:{}}
		// with modifiers already applied
		// args will be an object for parameters defined

		record.name = args.name;
		record.srvpriority = args.priority;
		record.srvweight = args.srvweight;
		record.srvport = args.srvport;
		record.target = args.target;

		return record;
	},
	// example of number modifier parsing (eg. in CAA)
	modifierNumber: function(record, value){
		record.caaflags |= value;
		return record;
	}
});
```

Record could implement `modifierNumber` and `modifierString` `modifierBoolean` (basically all primitive types but null/undefined) to support custom record modifier like what MX/CAA are using now. The default handler for those options will log "Warning: Number/String/Boolean modifier (value) is not supported. Skipping"

Is there any use case that I missed?

**Comment by captncraig** [comment]
I was thinking even simpler. Have addRecord return an object, and SRV just adds some fields to it after it finishes.

**Comment by whs** [comment]
That still retain the special cases of number modifier parsing for CAA/MX. As addRecord will warn on unknown modifier, handling the number modifiers in the CAA function will also trigger the warning.

**Comment by tlimoncelli** [comment]
Fixed in #157 and #163
